internal/kafka/internal/workers/clusters_mgr.go

```diff
@@ -558,17 +566,20 @@ func (c *ClusterManager) reconcileProvisionedCluster(cluster api.Cluster) error
 		return errors.WithMessagef(syncSetErr, "failed to reconcile cluster %s SyncSet: %s", cluster.ClusterID, syncSetErr.Error())
 	}
 
-	// Addon installation step
-	// TODO this is currently the responsible of setting the status of the cluster
-	// and it is setting it to a different value depending on the addon being
-	// installed. The logic to set the status of the cluster should probably done
-	// independently of the installation of the addon, and it should use the
-	// result of the addon/s reconciliation to set the status of the cluster.
-	addOnErr := c.reconcileAddonOperator(cluster)
+	addonsReconciled, addOnErr := c.reconcileAddonOperator(cluster)
```

Comment:
The logic of reconcileAddonOperator has been updated. Before, the update of the cluster status was included in that method. That has been extracted to make it independent from the reconcileAddonOperator method